Commonalities/documentation/API-design-guidelines.md

```diff
@@ -1021,3 +1027,180 @@ Finally, this part describes the OAuth security applied to the API. This spec is
+
+Providing this capability is optionnal for any CAMARA APIs depending on UC requirements
+
+If this capability is present in CAMARA API, following attributes **must** be used in the POST request for the managed entity:
+
+| attribute name | type | attribute description | cardinality |
+| ----- |	-----  |	 -----  | -----  | 
+| notificationUrl | string |callback address where the notification must be POST-ed | mandatory |
+| notificationAuthToken | string | authentification token for callback API | optional |
+
+
+**Resource-based subscription**
+
+A resource-based subscription is is subscription managed as a resource. A dedicated endpoint is provided to request subscription creation.  As this subscription is managed API resource, it is identified and operations to search, retrieve and delete them must be provided.
+
+Note: It is perfectly valid for a CAMARA API to have several subscriptions endpoints when distinct events are managed (for example in Device status API we manage distinct subscription for roaming status and reachability status event)
+
+In order to ease developer adoption, the pattern for Resource-based subscription must be the consistent for all API providing this feature.
+
+4 operations must be defined:
+
+| operation | path | description |
+| ----- |	-----  |	 -----  | 
+| POST | `/subscriptions/<subscriptionName>` | Operation to request a subscription.  eg. POST /subscriptions/roamingStatus    |
+| GET | `/subscriptions/` | Operation to retrieve a list of subscriptions - could be an empty list.  eg. `GET /subscriptions?type=ROAMING_STATUS&ExpireTime.lt=2023-03-17` |
+| GET | `/subscriptions/{subscriptionId}` | Operation to retrieve a subscription |
+| DELETE | `/subscriptions/{subscriptionId}` | Operation to delete a subscription |
+
+Following table provides subscriptions attributes
+
+| name | type | attribute description | cardinality |
+| ----- |	-----  |	 -----  |  -----  | 
+| notificationUrl | string | callback address where the notification must be POST-ed | mandatory |
+| notificationAuthToken | string | authentification token for callback API | optional |
+| type | string | type of notification subscribed. This attribute must not be present in the POST request as it is explicitly provided in the path| mandatory in server response |
+| subscriptionId | string | Identifier of the subscription - This attribute must not be present in the POST request as it is provided by API server | mandatory in server response |
+| subscriptionExpireTime | string - datetime| Date when the subscription should end. Provided by API requester | optional |
```

Comment:
What would the Use Case to delegate in the API consumer the settlement of the expirationTime?

Reply:
This was a request defined in Device location API by @NoelWirzius (https://github.com/camaraproject/DeviceLocation/issues/18).
ExpireTime to set how long the subscription should be active

This request makes additional work on server while while the DELETE operation will allow to perhaps simplify this. Perhaps we can have Noel feedback on this?

Reply:
Hey both, 

Yes the idea was to offer the developer an option to set a "Time window" for subscription so maybe just the next month. But to simplify it we can also work with no ExpireTime and just start with Create and a manual delete. 

@akoshunyadi  @rartych

Reply:
Probably here we would need to indicate that when subscriptionExpireTime, the Server may reject the suscription if the period requested do not comply with Telco Operator policies (i.e. to avoid unlimited time subscriptions)

Reply:
Added !